client/src/main/java/org/asynchttpclient/netty/ssl/DefaultSslEngineFactory.java

```diff
@@ -26,13 +26,11 @@
 
 public class DefaultSslEngineFactory extends SslEngineFactoryBase {
 
-    private final SslContext sslContext;
+    private volatile SslContext sslContext;
```

Comment:
Yeah, I have been trying to reject this FactoryFactory idea since yesterday and come with another solution. I've failed so far :(